java-manta-client-unshaded/src/main/java/com/joyent/manta/client/MantaClient.java

```diff
@@ -1237,26 +1237,36 @@ public MantaObjectResponse put(final String path,
     }
 
     /**
-     * Creates an OutputStream that wraps a PUT request to Manta. Try to avoid using this
+     * <p>Creates an OutputStream that wraps a PUT request to Manta. Try to avoid using this
      * to add data to Manta because it requires an additional thread to be started in order
      * to upload using an {@link java.io.OutputStream}. Additionally, if you do not close()
-     * the stream, the data will not be uploaded.
+     * the stream, the data will not be uploaded.</p>
      *
-     * @param path     The fully qualified path of the object. i.e. /user/stor/foo/bar/baz
+     * <p>NOTE: Use of this method should be discouraged because it uses another thread
+     * and it uses reflection in order to provide an {@link java.io.OutputStream} implementation.
+     * It is provided for users who have no choice but to use an {@link java.io.OutputStream} to
+     * write data to Manta due to constraints in their object model.</p>
+     *
+     * @param path The fully qualified path of the object. i.e. /user/stor/foo/bar/baz
      * @return A OutputStream that allows for directly uploading to Manta
      */
     public MantaObjectOutputStream putAsOutputStream(final String path) {
         return putAsOutputStream(path, null, null);
     }
 
     /**
-     * Creates an OutputStream that wraps a PUT request to Manta. Try to avoid using this
+     * <p>Creates an OutputStream that wraps a PUT request to Manta. Try to avoid using this
      * to add data to Manta because it requires an additional thread to be started in order
      * to upload using an {@link java.io.OutputStream}. Additionally, if you do not close()
-     * the stream, the data will not be uploaded.
+     * the stream, the data will not be uploaded.</p>
      *
-     * @param path     The fully qualified path of the object. i.e. /user/stor/foo/bar/baz
-     * @param headers  optional HTTP headers to include when copying the object
+     * <p>NOTE: Use of this method should be discouraged because it uses another thread
```

Comment:
Can we say anything more concrete to users besides threads+reflection == spooky == bad?  Will I deadlock? Is it super slow? Will my data get eaten?

Should we deprecate these methods?   Is an implementation without threads/reflection feasible?